pkg/api/types.go

```diff
@@ -708,6 +710,11 @@ func (a *AgentPoolProfile) IsVirtualMachineScaleSets() bool {
 	return a.AvailabilityProfile == VirtualMachineScaleSets
 }
 
+// IsLowPriorityScaleSet returns true if the VMSS is Low Priority
+func (a *AgentPoolProfile) IsLowPriorityScaleSet() bool {
+	return a.ScaleSetPriority == ScaleSetPriorityLow
```

Comment:
Should we also check that `availabilityProfile == VirtualMachineScaleSets ` here? Another option would be to add validation to ensure that the priority is empty when  `availabilityProfile != VirtualMachineScaleSets `. Depends on whether we want the user to be able to specify a priority even if they aren't using VMSS (and ignore that priority) or if we prefer strict checking that priority is only specified when VMSS is used (which means we can always use it if it's specified). What do you think?

Reply:
Thank you for catching this! I think the former check works well here and I've just added it.